pkg/fileutil/readdir_windows.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package fileutil
+
+import (
+	"os"
+	"path/filepath"
+	"sort"
+)
+
+// ReadDir is work-around for Windows.
+// TODO: use the same Readdir as in linux.
+// os.File.Readdirnames uses syscall.FindNextFile in Windows,
+// and it sometimes returns outdated files (os.FileInfo == nil).
+// filepath.Walk also uses exactly the same Readdirnames, but it allows
+// to check if the file is nil.
+func ReadDir(dir string) ([]string, error) {
+	var fnames []string
+	fp1 := filepath.Base(dir)
+	visit := func(path string, f os.FileInfo, err error) error {
+		if f != nil {
+			fp2 := filepath.Base(filepath.Dir(path))
+			if fp1 == fp2 { // to skip walking sub-directories like Readdir
+				fnames = append(fnames, f.Name())
+			}
+		}
+		return nil
+	}
+	if err := filepath.Walk(dir, visit); err != nil {
```

Comment:
OK, I looked at the go sources to figure out what's happening here. It seems the main difference between `readdirnames`/`readdir` and `walk` is `walk` is `lstat`ing the files after it gets the names so there could be some race happening and this is masking it.

Is this so the PurgeFile tests pass or is something else failing too?